deployment/scripts/codeship_app_engine_deploy

```diff
@@ -109,7 +119,15 @@ fi
 
 gcloud auth activate-service-account --key-file <(echo "${CODESHIP_GOOGLE_APP_ENGINE_KEY_CONTENT}")
 
-gcloud app deploy --project "${CODESHIP_GOOGLE_APP_ENGINE_PROJECT}"
+case "${CODESHIP_GOOGLE_APP_ENGINE_DEPLOY_COMMAND}" in
+"gcloud")
+  gcloud app deploy --project "${CODESHIP_GOOGLE_APP_ENGINE_PROJECT}"
+  ;;
+"maven")
+  mvn appengine:deploy -Dapp.deploy.project="${CODESHIP_GOOGLE_APP_ENGINE_PROJECT}"
+  ;;
+esac
```

Comment:
What's the reasoning to check the `CODESHIP_GOOGLE_APP_ENGINE_DEPLOY_COMMAND` variable in lines 101 to 105 and not have a default case here, which handles other cases and fails the command?

Reply:
Pretty much just wanted to have all the validation done before I actually execute anything. Just a habit in writing code I guess.